### PR TITLE
Dint 910 & DINT 911 - Round prices for productListItems for AEP events

### PR DESCRIPTION
**Pull request by benjaminkalk**
## Related Issue

https://jira.corp.adobe.com/browse/DINT-910
## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

-   [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [ ] I have read the **CONTRIBUTING** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/storefront-events-collector/src/utils/aep/productListItems.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ const createProductListItems = (
                 quantity: productListItemFromCustomContext?.quantity || Number(item.quantity),
                 priceTotal:
                     productListItemFromCustomContext?.priceTotal ||
-                    (Number(item.pricing?.regularPrice) || 0) * Number(item.quantity),
+                    Number(((Number(item.pricing?.regularPrice) || 0) * Number(item.quantity)).toFixed(2)),
```
Comment (grahamcrackers) [inline packages/storefront-events-collector/src/utils/aep/productListItems.ts:38]: I feel like a lot of context is lost here. I would move this parsing logic out into a function to make it more testable and then apply that function to any number values that need it.

File: packages/storefront-events-collector/src/utils/aep/productListItems.ts [hunk 2]
```diff
@@ -60,7 +60,9 @@ const createProductListItems = (
                 name: productListItemFromCustomContext?.name || item.product?.name,
                 quantity: productListItemFromCustomContext?.quantity || item.quantity,
                 priceTotal:
-                    productListItemFromCustomContext?.priceTotal || item.prices?.price?.value * item.quantity || 0,
+                    productListItemFromCustomContext?.priceTotal ||
+                    Number((item.prices?.price?.value * item.quantity).toFixed(2)) ||
+                    0,
```
Comment (grahamcrackers) [inline packages/storefront-events-collector/src/utils/aep/productListItems.ts:65]: same as the previous comment. This just feels a little kludgy.

**Comment by grahamcrackers** [comment]
I'm curious with these fixes, when the default currency is set to `EUR`, is the issue with the actual formatting, or is there a discrepency between using `,` and `.` as currency seperators.

**Comment by benjaminkalk** [comment]
> I'm curious with these fixes, when the default currency is set to `EUR`, is the issue with the actual formatting, or is there a discrepency between using `,` and `.` as currency seperators.

The problem is doing float math causing floating precision errors.  Also, magento core stores numbers to 4 decimals, but we are being required to have just the two decimal places.

**Review by NadiyaS — APPROVED** [review]

**Pull request merged**